### PR TITLE
Update for Xcode 8 & Swift 3.0

### PR DESCRIPTION
**Pull request by wmcbain**
Contains updates for Xcode 8 & Swift 3.0 compatibility. Simply updated the project with the conversion tool.

### CODE REVIEW

**Comment by kevin-hirsch** [comment]
Hello @wmcbain, thanks for the help!
I will get a quick look to the changes before merging your PR.

**Comment by kevin-hirsch** [comment]
Okay, here's the list (for me) of things left to do before we can merge this PR:

- [ ] change all `UIColor.colorName` using the shorter syntax to `.colorName` (e.g. `.red`, `.clear`) when `UIColor` is implicit.
- [ ] The `show()` function should be preceded by the `@discardableResult` attribute, because the returned value should be retrieved only when necessary. (Following the project guidelines, attributes should be on a single line before the function signature like for `@objc`s in code)

To embrace Swift 3, we should also change some API's according to the new Swift guidelines:
- [ ] `keyboardFrameFromNotification(_:)` to `keyboardFrame(fromNotification:)`
- [ ] `constraintWidthToBoundsIfNecessary(_:in:)` to `constraintWidthToBoundsIfNecessary(layout:in:)`
- [ ] `constraintWidthToFittingSizeIfNecessary(_:)` to `constraintWidthToFittingSizeIfNecessary(layout:)`
- [ ] `selectRowAtIndex(_:)` to `selectRow(at:)`
- [ ] `deselectRowAtIndexPath(_:)` to `deselectRow(at:)`

That would be perfect! 👌
What do you think @wmcbain?

File: DropDown/src/DropDown.swift [hunk 1]
```diff
@@ -7,6 +7,26 @@
 //
 
 import UIKit
+fileprivate func < <T : Comparable>(lhs: T?, rhs: T?) -> Bool {
```
Comment (kevin-hirsch) [inline DropDown/src/DropDown.swift]: These two functions should be removed. After investigation, there were added only for one use of operator `>` with an optional at line [570](https://github.com/AssistoLab/DropDown/blob/master/DropDown/src/DropDown.swift#L570):
```swift
barButtonItemCondition: if let _ = anchorView as? UIBarButtonItem {
	let isRightBarButtonItem = anchorView?.plainView.frame.minX > window.frame.midX

	guard isRightBarButtonItem else { break barButtonItemCondition }

	let width = self.width ?? fittingWidth()
	let anchorViewWidth = anchorView?.plainView.frame.width ?? 0
	let x = -(width - anchorViewWidth)

	bottomOffset = CGPoint(x: x, y: 0)
}
```

Which should actually be changed to:
```swift
barButtonItemCondition: if let anchorView = anchorView as? UIBarButtonItem {
	let isRightBarButtonItem = anchorView.plainView.frame.minX > window.frame.midX

	guard isRightBarButtonItem else { break barButtonItemCondition }

	let width = self.width ?? fittingWidth()
	let anchorViewWidth = anchorView.plainView.frame.width
	let x = -(width - anchorViewWidth)

	bottomOffset = CGPoint(x: x, y: 0)
}
```

**Comment by NemOry** [comment]
Can't wait for Swift3 version. Thank you

**Comment by wmcbain** [comment]
Sounds good to me. 👍

**Comment by kevin-hirsch** [comment]
@wmcbain You think you finish this before the weekend? 😀
Or you prefer that I do the changes?

**Comment by wmcbain** [comment]
I'll give it a go, be on the lookout for the commit tonight.

**Comment by kevin-hirsch** [comment]
One last thing to do will be to change the single mention of `selectRowAtIndex` in *README.md* for `selectRow(at:)` 😉

File: DropDown/src/DropDown.swift [hunk 2]
```diff
@@ -861,20 +842,20 @@ extension DropDown {
 	}
 
 	/// (Pre)selects a row at a certain index.
-	public func selectRowAtIndex(_ index: Index?) {
+	public func selectRow(atIndex index: Index?) {
```
Comment (kevin-hirsch) [inline DropDown/src/DropDown.swift]: Wouldn't `selectRow(at index: Index?)` be nicer? Like the Swift 3 API for `UITableView` which have [`selectRow(at:animated:scrollPosition:)`](https://developer.apple.com/reference/uikit/uitableview/1614875-selectrow)

File: DropDown/src/DropDown.swift [hunk 3]
```diff
@@ -861,20 +842,20 @@ extension DropDown {
...
 		} else {
-			deselectRowAtIndexPath(selectedRowIndex)
+            deselectRow(atIndex: selectedRowIndex)
 		}
 
 		selectedRowIndex = index
 	}
 
-	public func deselectRowAtIndexPath(_ index: Index?) {
+	public func deselectRow(atIndex index: Index?) {
```
Comment (kevin-hirsch) [inline DropDown/src/DropDown.swift]: Same here: `deselectRow(at index: Index?)`?

**Comment by wmcbain** [comment]
I'm unsure why Github is showing the indentation off. I have a 4 space tab set in my Xcode preferences and the indentation level is correct in the editor. Might just be a fluke in Github's UI? Or is this appearing in the project locally for you as well?

File: DropDown/helpers/DPDKeyboardListener.swift [hunk 4]
```diff
@@ -52,16 +52,16 @@ extension KeyboardListener {
...
 	
 	@objc
 	fileprivate func keyboardWillHide(_ notification: Notification) {
 		isVisible = false
-		keyboardFrame = keyboardFrameFromNotification(notification)
+		keyboardFrame = keyboardFrame(fromNotification: notification)
 	}
 	
-	fileprivate func keyboardFrameFromNotification(_ notification: Notification) -> CGRect {
+    fileprivate func keyboardFrame(fromNotification notification: Notification) -> CGRect {
```
Comment (kevin-hirsch) [inline DropDown/helpers/DPDKeyboardListener.swift]: I do see in my Xcode the indent with 4 actual spaces.

File: DropDown/src/DropDown.swift [hunk 5]
```diff
@@ -310,7 +290,7 @@ public final class DropDown: UIView {
 	*/
 	public var dataSource = [String]() {
 		didSet {
-			deselectRowAtIndexPath(selectedRowIndex)
+            deselectRow(atIndex: selectedRowIndex)
```
Comment (kevin-hirsch) [inline DropDown/src/DropDown.swift]: Here also indents are with spaces.

File: DropDown/src/DropDown.swift [hunk 6]
```diff
@@ -586,17 +566,17 @@ extension DropDown {
...
-
-			guard isRightBarButtonItem else { break barButtonItemCondition }
-
-			let width = self.width ?? fittingWidth()
-			let anchorViewWidth = anchorView?.plainView.frame.width ?? 0
-			let x = -(width - anchorViewWidth)
-
-			bottomOffset = CGPoint(x: x, y: 0)
-		}
+        barButtonItemCondition: if let anchorView = anchorView as? UIBarButtonItem {
```
Comment (kevin-hirsch) [inline DropDown/src/DropDown.swift]: Here also indents are with spaces (also in following lines).

File: DropDown/src/DropDown.swift [hunk 7]
```diff
@@ -624,8 +604,8 @@ extension DropDown {
 			}
 		}
 		
-		constraintWidthToFittingSizeIfNecessary(&layout)
-		constraintWidthToBoundsIfNecessary(&layout, in: window)
+        constraintWidthToFittingSizeIfNecessary(layout: &layout)
```
Comment (kevin-hirsch) [inline DropDown/src/DropDown.swift]: Same here for both lines: indents with spaces.

File: DropDown/src/DropDown.swift [hunk 8]
```diff
@@ -795,7 +776,7 @@ extension DropDown {
 			},
 			completion: nil)
 
-		selectRowAtIndex(selectedRowIndex)
+        selectRow(atIndex: selectedRowIndex)
```
Comment (kevin-hirsch) [inline DropDown/src/DropDown.swift]: Same here for both lines: indents with spaces.

File: DropDown/src/DropDown.swift [hunk 9]
```diff
@@ -861,20 +842,20 @@ extension DropDown {
...
-	public func selectRowAtIndex(_ index: Index?) {
+	public func selectRow(atIndex index: Index?) {
 		if let index = index {
 			tableView.selectRow(
 				at: IndexPath(row: index, section: 0),
 				animated: false,
 				scrollPosition: .middle)
 		} else {
-			deselectRowAtIndexPath(selectedRowIndex)
+            deselectRow(atIndex: selectedRowIndex)
```
Comment (kevin-hirsch) [inline DropDown/src/DropDown.swift]: Same here for both lines: indents with spaces.

File: DropDown/src/DropDown.swift [hunk 10]
```diff
@@ -948,7 +929,7 @@ extension DropDown: UITableViewDataSource, UITableViewDelegate {
 
 		if let _ = anchorView as? UIBarButtonItem {
 			// DropDown's from UIBarButtonItem are menus so we deselect the selected menu right after selection
-			deselectRowAtIndexPath(selectedRowIndex)
+            deselectRow(atIndex: selectedRowIndex)
```
Comment (kevin-hirsch) [inline DropDown/src/DropDown.swift]: Same here for both lines: indents with spaces.

**Comment by wmcbain** [comment]
Okay I know this seems silly but you may have to make those changes then. I have my preferences set to your indentation scheme, they appear correct in my editor, and reindenting doesn't cause a diff that would be the fix you're looking for. No idea why I'm seeing this behavior.

**Comment by wmcbain** [comment]
screen shot 2016-09-16 at 6 52 59 am

**Comment by kevin-hirsch** [comment]
Alright no problem, I'll do it! 👍

The configuration I have is different here:
_Prefer indent using: **Tabs**_

Did you check my comments about `selectRow(at:)` and `deselectRow(at:)`? :)

**Comment by wmcbain** [comment]
```The indentation in the project is using tabs (of 4 spaces).```

I read this a bit wrong. I can make the changes described then for both indenting and the row selection methods.

**Comment by wmcbain** [comment]
Think we should be all set on this.

**Comment by kevin-hirsch** [comment]
Thanks a lot @wmcbain! 👍🍾

**Comment by wmcbain** [comment]
Hey as a suggestion, maybe a version bump and a podspec update for versioning Cocoapods users?

**Comment by kevin-hirsch** [comment]
I'm updating the _README.md_ right now to Swift 3 and then I will provide a new version of `DropDown` through CocoaPods and Carthage 😉

**Comment by kevin-hirsch** [comment]
I did create `DropDown` version 2 🎉
- [x] REAMDE.md updated
- [x] CocoaPods version 2 available
- [x] Carthage version 2 available

**Pull request merged**